File: 1-two-sum.scala
//https://leetcode.com/problems/two-sum/description/?envType=daily-question&envId=2025-07-19 
import scala.util.control.Breaks._

object Solution {
  def twoSum(nums: Array[Int], target: Int): Array[Int] = {
    var result: Array[Int] = Array()

    breakable {
      for (i <- nums.indices; j <- nums.indices) {
        if (i != j && nums(i) + nums(j) == target) {
          result = Array(i, j)
          break()
        }
      }
    }
    result
  }
}

File: 1-two-sum.ts
/* https://leetcode.com/problems/two-sum/description/?envType=daily-question&envId=2025-07-19 */
function twoSum(nums: number[], target: number): number[] {
  const map = new Map();
  for (let i = 0; i < nums.length; i++) {
    const number = nums[i];
    const remaining = target - number;
    if (map.has(remaining)) return [map.get(remaining), i];
    map.set(number, i);
  }
  return [];
}
